### PR TITLE
Hit the API for storing and getting scores

### PR DESCRIPTION
**Pull request by Asiimwe-Mark**
I used the leaderBoard API and the fetch API to capture scores from the player and store it and then retrieve it when needed.
I followed the following guidelines

Create a new game with the name of your choice by using the API.
Make sure that you saved the ID of your game that will be returned by API.
Implement the "Refresh" button that will get all scores for a game created by you from the API (receiving data from the API and parsing the JSON).
Implement the form "Submit" button that will save a score for a game created by you (sending data to the API).
Use arrow functions instead of the function keyword.
Use async and await JavaScript features to consume the API.
No styling is required.

### CODE REVIEW

File: src/Modules/board.js [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
-  refreshBtn.addEventListener('click', (e) => {
-    e.preventDefault();
-    const scor = JSON.parse(localStorage.getItem('Scores') || []);
-    let boardHtml = '';
-    if (scor) {
-      Scores = scor;
-      Scores.forEach((player) => {
+async function ScoreBoard() {
+  await fetch('https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/:OsIdfjVZGF6rZmZWBTi6/scores/')
+    .then((res) => res.json())
```
Comment (Olamarx) [inline src/Modules/board.js]: - [ ] I will encourage you to take advantage of the arrow function as the function keyword is professionally not recommendable and it also helps javascript hoists in the runtime.

File: src/Modules/createGame.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+async function createGame() {
+  await fetch('https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/', {
+    method: 'POST',
+    headers: { 'content-Type': 'application/json' },
```
Comment (Olamarx) [inline src/Modules/createGame.js]: - [ ] Same thing here, kindly apply it to all your functions.

File: src/Modules/input.js [hunk 3]
```diff
@@ -1,20 +1,11 @@
...
-    const scores = {
-      Name: name,
-      Score: +score,
-    };
-    Scores.push(scores);
-    document.querySelector('#name').value = '';
-    document.querySelector('#score').value = '';
+async function getInput() {
+  const name = document.querySelector('#name').value.trim();
+  const score = document.querySelector('#score').value.trim();
```
Comment (Olamarx) [inline src/Modules/input.js]: - [ ] Remember this too

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @As1imwe-Mark, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Async and Await and fetch method ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: dist/index.html [hunk 4]
```diff
@@ -13,7 +13,27 @@ <h1>Leaderboard</h1>
...
+                      </tr>
+                      <tr>
+                        <td>Name:125</td>
+                      </tr>
+                      <tr>
+                        <td>Name:77</td>
+                      </tr>
+                      <tr>
+                        <td>Name:42</td>
+                      </tr> -->
```
Comment (shahadat3669) [inline dist/index.html:36]: - [ ] [Optional] Well done on converting all the functions into arrow functions! 👍 However, I noticed that you still have some commented code in your source code. Commented-out code is code that has been excluded or disabled from execution using special characters like "//" or "/* */".  Commented-out code takes up unnecessary space in your codebase and can make it harder to read and understand the active code. Commented-out code may cause confusion for other developers who are working on the project, as they might wonder if it's still relevant or intended to be used. 

- [ ] [Optional] To maintain a clean and organized codebase, I recommend removing all commented-out code that is no longer needed. If you need to keep a record of the old code for reference, consider using version control systems like Git to manage previous versions of your code.

- [ ] [Optional] By removing the commented-out code, you'll ensure that your codebase stays clear and concise, making it easier for you and your team to work on the project efficiently. Keep up the good work! 🚀🔧🧹

**Review by shahadat3669 — APPROVED** [review]
Hi @As1imwe-Mark , 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No linter Errors
✅ Descriptive PR Title and Description
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- [Optional] Great job on using HTML, CSS, and JavaScript in your project! 👏 Setting up linters for CSS and JavaScript is a step in the right direction to maintain code quality and consistency. However, I noticed that you haven't set up a linter for HTML. Using a linter for HTML can be very beneficial as it helps identify and fix errors, maintain code style, and ensure adherence to best practices. To set up a linter for HTML, follow the [linters documentation](https://github.com/microverseinc/linters-config/tree/master). By setting up an HTML linter, you'll further enhance the overall code quality and maintainability of your project. Keep up the good work and continue improving your development practices! 🌟🚀🔧

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**